src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -276,8 +279,64 @@ export class ObjectExplorerService implements IObjectExplorerService {
 	private async handleSessionCreated(session: azdata.ObjectExplorerSession): Promise<void> {
 		let connection: ConnectionProfile | undefined = undefined;
 		let errorMessage: string | undefined = undefined;
-		if (this._sessions[session.sessionId!]) {
-			connection = this._sessions[session.sessionId!].connection;
+		const sessionId = session.sessionId;
+
+		/**
+		 * Sometimes ads recieves handleSessionCreated from providers before the createNewSession response is recieved.
+		 * We need to wait while the createNewSession response is recieved and the session map contains the session before we can continue.
+		 */
+		await new Promise<void>((resolve, reject) => {
+			let count = 0;
+			const waitingForSessionId = () => {
+				const session = this._sessions[sessionId];
+				if (session) {
+					clearTimeout(timerId);
+					resolve();
+				} else if (count > this.getObjectExplorerTimeout()) {
+					clearTimeout(timerId);
+					reject(new Error(
+						nls.localize(
+							'objectExplorerMissingSession',
+							'Timed out while waiting for createNewSession response from the provider for this created session `{0}`', sessionId
```

Comment:
We don't really use backticks for quotes anywhere else so that should be avoided - and here the session ID isn't going to contain any spaces so is fine to just have not quoted I think. 

```suggestion
							'Timed out while waiting for createNewSession response from the provider for this created session {0}', sessionId
```